mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+from .util import midi_to_hz, hz_to_midi
+
+
+def __expand_limits(ax, limits, which='x'):
+    '''Helper function to expand axis limits'''
+
+    if which == 'x':
+        getter, setter = ax.get_xlim, ax.set_xlim
+    elif which == 'y':
+        getter, setter = ax.get_ylim, ax.set_ylim
+    else:
+        raise ValueError('invalid axis: {}'.format(which))
+
+    old_lims = getter()
+    new_lims = list(limits)
+
+    if np.isfinite(old_lims[0]):
+        new_lims[0] = min(old_lims[0], limits[0])
+
+    if np.isfinite(old_lims[1]):
+        new_lims[1] = max(old_lims[1], limits[1])
+
+    setter(new_lims)
+
+
+def segments(intervals, labels, base=None, height=None, text=False,
+             text_kw=None, ax=None, **kwargs):
+    '''Plot a segmentation as a set of disjoint rectangles.
+
+    Parameters
+    ----------
+    intervals : np.ndarray, shape=(n, 2)
+        segment intervals, in the format returned by
+        :func:`mir_eval.io.load_intervals` or
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    labels : list, shape=(n,)
+        reference segment labels, in the format returned by
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    base : number
+        The vertical position of the base of the rectangles.
+        By default, this will be the bottom of the plot.
+
+    height : number
+        The height of the base of the rectangles.
+        By default, this will be the top of the plot (minus `base`).
+
+    text : bool
+        If true, each segment's label is displayed in its
+        upper-left corner
+
+    text_kw : dict
+        If `text==True`, the properties of the text
+        object can be specified here.
+        See `matplotlib.pyplot.text` for valid parameters
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the segmentation.
+        If none is provided, a new set of axes is created.
+
+    kwargs
+        Additional keyword arguments to pass to
+        `matplotlib.patches.Rectangle`.
+
+    Returns
+    -------
+    ax : matplotlib.pyplot.axes._subplots.AxesSubplot
+        A handle to the (possibly constructed) plot axes
+    '''
+    if text_kw is None:
+        text_kw = dict(va='top',
+                       clip_on=True,
+                       bbox=dict(boxstyle='round', facecolor='white'))
+
+    seg_def_style = dict(linewidth=1)
+
+    if ax is None:
+        # Create a new axis
+        ax = plt.gca()
+        ax.set_ylim([0, 1])
+
+    # Infer height
+    if base is None:
+        base = ax.get_ylim()[0]
+
+    if height is None:
+        height = ax.get_ylim()[1]
+
+    cycler = ax._get_patches_for_fill.prop_cycler
+
+    seg_map = dict()
+
+    for lab in labels:
+        if lab in seg_map:
+            continue
+
+        style = next(cycler)
+        seg_map[lab] = seg_def_style.copy()
+        seg_map[lab].update(style)
+        seg_map[lab]['facecolor'] = seg_map[lab].pop('color')
+        seg_map[lab].update(kwargs)
+        seg_map[lab]['label'] = lab
```

Comment:
No, it's for legend entries.  If you want to use a legend instead of the embedded text boxes, this is how you set that attribute.  We pop it after drawing the first one so you don't get redundant legend entries.